workflow/artifacts/common/common.go

```diff
@@ -1,12 +1,18 @@
 package common
 
-import "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+import (
+	"io"
+
+	"github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+)
 
 // ArtifactDriver is the interface for loading and saving of artifacts
 type ArtifactDriver interface {
 	// Load accepts an artifact source URL and places it at specified path
 	Load(inputArtifact *v1alpha1.Artifact, path string) error
 
+	OpenStream(inputArtifact *v1alpha1.Artifact) (io.ReadCloser, error)
```

Comment:
The qualifier "input" is not needed,  this can be any kind of artifact, I'd just change to "a" in fact. You should make this change in each driver.

I think we should add a comment saying the following:

```golang
// OpenStream opens an artifact for reading. If the artifact is a file, then the file should be opened.
// If the artifact is a directory, the driver may return that as a tarball. OpenStream is intended to be efficient, 
// so implementations should minimise usage of disk, CPU and memory. 
// Implementations must not implement retry mechanisms. This will be handled by the client, so would result in O(nm) cost.
```

Reply:
is that an error in the Load method above too then?

Reply:
no need - just for this